src/cls/rbd/cls_rbd_types.h

```diff
@@ -133,6 +133,7 @@ inline void decode(MirrorImageStatusState &state, bufferlist::iterator& it)
 
 struct MirrorImageStatus {
   MirrorImageStatus() {}
+  ~MirrorImageStatus() {}
```

Comment:
"If you have a non-virtual destructor, have an object from the derived class, and delete it using the base class pointer -- it just doesn't invoke the derived class destructor (i.e. memory leak or broken functionality)."

So isn't that an argument for declaring this particular destructor virtual?